ghcide/src/Development/IDE/Plugin/Completions/Logic.hs

```diff
@@ -890,3 +929,33 @@ mergeListsBy cmp all_lists = merge_lists all_lists
         []     -> []
         [xs]   -> xs
         lists' -> merge_lists lists'
+
+
+getCompletionPrefix :: (Monad m) => Position -> VFS.VirtualFile -> m (Maybe PosPrefixInfo)
+getCompletionPrefix pos@(Position l c) (VFS.VirtualFile _ _ ropetext) =
+      return $ Just $ fromMaybe (PosPrefixInfo "" "" "" pos) $ do -- Maybe monad
+        let headMaybe []    = Nothing
+            headMaybe (x:_) = Just x
+            lastMaybe []     = Nothing
+            lastMaybe [x]    = Just x
+            lastMaybe (_:xs) = lastMaybe xs
+
+        curLine <- headMaybe $ T.lines $ Rope.toText
```

Comment:
what are you trying to do? worth explaining, much of this is not obvious

Reply:
Also this is different from the `lsp` version now, what changed?

Reply:
I'll document in comments

Reply:
But the reason for the change was different rope libraries, though it looks like the versions are the same now.